railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -80,7 +80,8 @@ def self.plugin_rails_init(options)
       Minitest.backtrace_filter = ::Rails.backtrace_cleaner if ::Rails.respond_to?(:backtrace_cleaner)
     end
 
-    self.reporter.reporters.clear # Replace progress reporter for colors.
+    # Replace progress reporter for colors.
+    self.reporter.reporters.delete_if { |reporter| reporter.is_a?(SummaryReporter) || reporter.is_a?(ProgressReporter) }
```

Comment:
Since all of the custom reporters will not mean that conflict of reporter of the rails, I think it should not be delete all.
In my project, I am using reporters that extends `Minitest::StatisticsReporter`, they do not conflict with the reporter of the rails.